pkg/provisioner/ironic/ironic.go

```diff
@@ -94,14 +95,12 @@ func init() {
 	}
 
 	deployKernelURL = os.Getenv("DEPLOY_KERNEL_URL")
-	if deployKernelURL == "" {
-		fmt.Fprintf(os.Stderr, "Cannot start: No DEPLOY_KERNEL_URL variable set\n")
-		os.Exit(1)
-	}
 	deployRamdiskURL = os.Getenv("DEPLOY_RAMDISK_URL")
-	if deployRamdiskURL == "" {
-		fmt.Fprintf(os.Stderr, "Cannot start: No DEPLOY_RAMDISK_URL variable set\n")
+	deployISOURL = os.Getenv("DEPLOY_ISO_URL")
+	if deployISOURL == "" && (deployKernelURL == "" || deployRamdiskURL == "") {
+		fmt.Fprintf(os.Stderr, "Cannot start: Either DEPLOY_ISO_URL or DEPLOY_KERNEL_URL and DEPLOY_RAMDISK_URL must be set\n")
```

Comment:
I tried to read this error message from the point of view of the user, and it looks like a little bit tricky. Wondering if it's worh splitting the checks, validating first `DEPLOY_RAMDISK_URL` and, if not set, validating `DEPLOY_KERNEL_URL` and `DEPLOY_RAMDISK_URL` as usual

Reply:
Not sure I get what you mean. Could you write pseudo-code for your proposal?

Reply:
Sorry, just noticed a typo in my comment, anyhow by reviewing the changes I realized that some points are still unclear to me, in particular which are the allowed/valid combinations? From the code it seems that `ISO`, `ISO/RAMDISK`, `ISO/KERNEL`,`ISO/RAMDISK/KERNEL` and `KERNEL/RAMDISK` could pass the validation check (while ` `, `KERNEL` and `RAMDISK` will fail)?

Reply:
Classic operator precedence problem :laughing:

"Either DEPLOY_KERNEL_URL and DEPLOY_RAMDISK_URL or DEPLOY_ISO_URL must be set" would be unambiguous.